packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -48,6 +43,29 @@ abstract class MacOSApp extends ApplicationPackage {
     );
   }
 
+  static Future<MacOSApp> fromMacosProject(
```

Comment:
This doesn't need to return a `Future`.  It could also stay in the `factory` until this file is migrated to null safety.